### PR TITLE
implementing OMP threads for sample 2p1d and 3p1d

### PR DESCRIPTION
**Pull request by latessa**
I've marked this as a draft because the 3p1d execution is crashing for me.  But the original roch/improve_parton_sampler branch is also crashing the same way for the 3p1d execution, so perhaps there's a problem with my input.  The 2p1d execution is working for me and producing reproducible results with any number of threads used.

### CODE REVIEW

File: src/hadronization/HybridHadronization.cc [hunk 1]
```diff
@@ -573,13 +573,17 @@ void HybridHadronization::DoHadronization(vector<vector<shared_ptr<Parton>>>& sh
       surface.emplace_back(std::move(cell));
     }
 
+    auto start = std::chrono::high_resolution_clock::now();
```
Comment (Hendrik1704) [inline src/hadronization/HybridHadronization.cc:576]: The timing should not propagate into the final version.

File: src/hadronization/ThermPtnSampler.cc [hunk 2]
```diff
@@ -466,71 +471,80 @@ std::vector<double> ThermalPartonSampler::LorentzBoost(std::vector<double>& x, s
...
-				Plist.back()[1] = -3;
+				temp[1] = -3;
 			} else { // S
-				Plist.back()[1] = 3;
+				temp[1] = 3;
 			}
 		} else {
 			JSWARN << "This is not a valid quark input for SamplePartons()";
-			return;
+			// return;
```
Comment (Hendrik1704) [inline src/hadronization/ThermPtnSampler.cc:510]: This comment could be removed

**Review by Hendrik1704 — COMMENTED** [review]
I went through the code and have only minor comments. I did not run it.

File: src/hadronization/HybridHadronization.cc (hunk 1 above)
Comment (latessa) [reply src/hadronization/HybridHadronization.cc]: I removed these time commands.  Thanks.

File: src/hadronization/ThermPtnSampler.cc (hunk 2 above)
Comment (latessa) [reply src/hadronization/ThermPtnSampler.cc]: I removed this comment.

**Pull request merged**